libraries/Wire/src/Wire.h

```diff
@@ -34,8 +34,9 @@
 #endif
 #include "Stream.h"
 
+#define WIRE_HAS_BUFFER_SIZE    1
```

Comment:
Just for info, what is this define used for? Thanks

Reply:
This `#define` is common in Arduino Mainstream to allow other implementations to add or not a specific functionality in their code. 

Example: 
https://github.com/arduino/ArduinoCore-avr/blob/master/libraries/Wire/src/Wire.h#L31-L32
``` cpp
// WIRE_HAS_END means Wire has end()
#define WIRE_HAS_END 1
```

Reply:
If this is common in Arduino why not to have it :) maybe we can add `WIRE_HAS_END` as well :)

Reply:
If its common in Arduino mainstream why not to have it :) Maybe we can add the `#define WIRE_HAS_END 1` as well in this PR. What do you think?

Reply:
done